### PR TITLE
Improve the "pre commit" scripts to be more human friendly

### PR DESCRIPTION
**Pull request by efrisch**
Removed a typo in the shell version, added echo off to the bat. Made sure to use bash. Use printf instead of echo. Check that commands succeed. Exit on error. Don't continue.

## PR Checklist

- [ ] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [ ] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [ ] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [ ] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

- [ ] I've included a brief description and a good title of the proposed changes and how to test them.
- [ ] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [ ] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [ ] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [ ] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [ ] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes pyrsia/pyrsia#

This PR does... by accomplishing ...

### CODE REVIEW

File: pre-commit.bat [hunk 1]
```diff
@@ -1,9 +1,11 @@
-echo "Welcome to the Pyrsia local tests.\nThis might take sometime, donot interrupt it if the screen is blank."
-cargo clean;
-cargo install cargo-audit;
-cargo audit;
-cargo clippy;   
-rustup component add rustfmt;
-cargo fmt --check;
-cargo test --workspace;
+@echo off
+echo "Welcome to the Pyrsia local tests."
```
Comment (prince-chrismc) [inline pre-commit.bat:2]:
```suggestion
echo "Running Pyrsia pre-commit validation."
```

File: pre-commit.sh [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-cargo clean;
-cargo install cargo-audit;
-cargo audit;
-cargo clippy;   
-rustup component add rustfmt;
-cargo fmt --check;
-cargo "test --workspace";
-cargo build --workspace;
+#!/usr/bin/env bash
+printf "Welcome to the Pyrsia local tests.\nThis might take sometime, please do not interrupt if the screen is blank.\n"
```
Comment (prince-chrismc) [inline pre-commit.sh:2]:
```suggestion
printf "Running Pyrsia pre-commit validation.\nThis might take sometime, please do not interrupt if the screen is blank.\n"
```

File: pre-commit.bat [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+echo "This might take sometime, donot interrupt it if the screen is blank."
+cargo clean
```
Comment (prince-chrismc) [inline pre-commit.bat:4]:
```suggestion
if ["%~1"]==["clean"] (
  echo "Cleaning old build artifacts".
  cargo clean
)
```

**Review by prince-chrismc — COMMENTED** [review]
Two minor items

- keep batch inline with shell for clean functionality
- words

**Review by AbhijithGanesh — DISMISSED** [review]

File: pre-commit.sh [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+if [[ "$1" == "clean" ]] ; then
+	printf "Cleaning old build artifacts.\n"
+	cargo clean
+fi
```
[the remark this answers is not in the document]
Comment (efrisch) [reply pre-commit.sh:7]: @prince-chrismc I've done a 2nd pass on the DOS side. Please re-review when convenient.

**Review by AbhijithGanesh — APPROVED** [review]

**Comment by prince-chrismc** [comment]
@efrisch please fill in the template or leave a comment explaining why you did use it 🙏

**Review by prince-chrismc — APPROVED** [review]

**Review by betarelease — APPROVED** [review]
The only addition I may add is a 
set -x at the top of the bash script
so that it shows what is being run.

**Pull request merged**